electron/js/controller.js

```diff
@@ -351,52 +340,69 @@ function clear() {
 }
 
 // This handles file selection via the button
-document.getElementById('file-loader').addEventListener('change', handleFileSelect, false);
+document.getElementById("file-loader").addEventListener("change", handleFileSelect, false);
+
 function handleFileSelect(evt) {
-  var files = evt.target.files
+  var files = evt.target.files;
   if(!files || !files.length) return;
   for(var i = 0, f; i < files.length; i++) {
```

Comment:
this is actually an old bug, but we have set ourselves up for a scope trap here. Below we have an async call within a for loop, bad news!
not only that, but we don't really support loading multiple files. I'm pretty sure the file input only allows you to load 1 anyway, but this is kind of a trap for the future where we may want to let people run multiple datasets for some reason.